src/app/profile/payment-methods/payment-methods.component.js

```diff
@@ -0,0 +1,140 @@
+import angular from 'angular';
+import template from './payment-methods.tpl';
+import recurringGiftsComponent from './recurring-gifts/recurring-gifts.component';
+import profileService from 'common/services/api/profile.service.js';
+import paymentMethod from './payment-method/payment-method.component';
+import paymentMethodFormModal from 'common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+import paymentMethodDisplay from 'common/components/paymentMethods/paymentMethodDisplay.component';
+import sessionEnforcerService from 'common/services/session/sessionEnforcer.service';
+import {Roles} from 'common/services/session/session.service';
+
+class PaymentMethodsController {
+
+  /* @ngInject */
+  constructor($uibModal, profileService, sessionEnforcerService, $log, $timeout, $window, $location) {
+    this.log = $log;
+    this.$uibModal = $uibModal;
+    this.paymentMethod = 'bankAccount';
+    this.profileService = profileService;
+    this.loading = false;
+    this.submissionError = {error: ''};
+    this.successMessage = { show: false };
+    this.$timeout = $timeout;
+    this.$window = $window;
+    this.$location = $location;
+    this.sessionEnforcerService = sessionEnforcerService;
+  }
+
+  $onDestroy(){
+    if(this.paymentMethodFormModal) {
+      this.paymentMethodFormModal.close();
+    }
+  }
+
+  $onInit(){
+    this.enforcerId = this.sessionEnforcerService([Roles.registered], {
```

Comment:
sessionEnforcer has changed slightly with the latest PR #178. sessionEnforcer will now call the sign-in callback if the user is already signed in, meaning we don't need lines 47 - 49 anymore. sessionEnforcer will also enforce a donor session, which is required for profile pages. To enable this, add 'donor' as the third parameter to sessionEnforcerService.

See [Your Giving](https://github.com/CruGlobal/give-web/blob/master/src/app/profile/yourGiving/yourGiving.component.js#L45) for example.